### DISCUSSION TITLE
How to show invalid credentials or token expired message in case of multiple authentication enabled per api?

### DISCUSSION THREAD

**Discussion by amitsit06** [Kong/kong#7264] [Ideas and feature requests]
NOTE: GitHub issues are reserved for **bug reports only**. For anything else,
please join the conversation in Kong Nation https://discuss.konghq.com/.

Please read the CONTRIBUTING.md guidelines to learn on which channels you can
seek for help and ask general questions:

https://github.com/Kong/kong/blob/master/CONTRIBUTING.md#where-to-seek-for-help

### Summary

In kong, I have configured key-auth and JWT plugin on my api. In case of invalid credential or invalid token, request fallback to anonymous consumer having request-termination plugin. Hence, request terminates with status code and message configured in request-termination plugin.
Is there a way request-termination plugin terminates the request but still show validation messages coming out of key-auth or JWT plugin.
Issue I am facing is - how do I show invalid grant_type or token expired validations done by JWT plugin.

### Steps To Reproduce

1. create an api
2. enable key-auth and JWT plugin on the api
3. create a consumer, create key-auth credentials for it. Also, create RS256 algorithm based JWT credential for it.
4. create an anonymous consumer. Create request-termination plugin for it.
5. Specify Kong UUID of anonymous consumer in config.anonymous field value in key-auth and JWT plugin earlier enabled for the api.

### Additional Details & Logs

- Kong version 0.14.1

**Comment by bungle** [comment]
@amitsit06, currently you cannot. We have plans to rework plugins runloop in a future, and authentication is surely something that needs some reworking, and suggestion you made here, makes a sense, so thank you for reporting. Unfortunately at this point there is no answer to that.
